crates/extensions/tedge_config_manager/src/actor.rs

```diff
@@ -223,6 +228,36 @@ impl ConfigManagerActor {
         Ok(())
     }
 
+    fn create_tedge_url_for_config_operation(
+        &self,
+        topic: &Topic,
+    ) -> Result<String, EntityTopicError> {
+        let (device_name, operation_type, cmd_id) =
+            match self.config.mqtt_schema.entity_channel_of(topic) {
+                Ok((entity, Channel::Command { operation, cmd_id })) => {
+                    match entity.default_device_name() {
+                        Some(device_name) => (device_name.to_owned(), operation, cmd_id),
+                        None => {
+                            return Err(EntityTopicError::TopicId(
+                                tedge_api::mqtt_topics::TopicIdError::InvalidMqttTopic,
+                            ))
+                        }
+                    }
+                }
+
+                _ => {
+                    return Err(EntityTopicError::Channel(
+                        tedge_api::mqtt_topics::ChannelError::InvalidCategory(topic.name.clone()),
+                    ));
+                }
+            };
+
+        Ok(format!(
+            "http://{}/tedge/file-transfer/{}/{}/{}",
+            &self.config.tedge_http_host, device_name, operation_type, cmd_id
+        ))
```

Comment:
Actually `cmd_id` here contains both `config_type` and id of operation. `entity_channel_of` method scraps both information from topic under one variable, whereas in `tedge-mapper` `cmd_id` is an operation number directly taken from the payload. And we know that `cmd_id` does not contain `/`, otherwise it would be incorrect topic. Maybe I can use another name for that variable to not confuse it with `cmd_id` used in mapper.

Reply:
But your test indicates that `cmd_id` here is purely command ID, in the example, `local-3333`. 

```robot
    Publish and Verify Local Command
    ...    topic=te/device/main///cmd/config_snapshot/local-3333
    ...    payload={"status":"init","type":"tedge-configuration-plugin"}
    ...    expected_status=successful
    ...    c8y_fragment=c8y_UploadConfigFile

    ${messages}=    Should Have MQTT Messages
    ...    te/device/main///cmd/config_snapshot/local-3333
    ...    message_contains=http://${PARENT_IP}:8000/tedge/file-transfer/main/config_snapshot/local-3333
    ...    date_from=${timestamp}
```

Reply:
Ok yeah that makes sense, previously I put `tedge-configuration-plugin-1111` instead of `local-3333` and the resulting url was: `http://${PARENT_IP}:8000/tedge/file-transfer/main/config_snapshot/tedge-configuration-plugin-tedge-configuration-plugin-1111` but yeah it should be only ID in topic.

Reply:
Resolved by ef29c2d0e8c337c522c0642bc58c462959337c4c